src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAITextToImageClient.cs

```diff
@@ -0,0 +1,252 @@
+    {
+        _ = Throw.IfNull(originalImage);
+        _ = Throw.IfNull(originalImageFileName);
+        _ = Throw.IfNull(prompt);
+        _ = Throw.IfNull(options);
+
+        ImageEditOptions openAIOptions = ToOpenAIImageEditOptions(options);
+        ImageClient imageClient = GetImageClient(options);
+
+        GeneratedImageCollection result = await imageClient.GenerateImageEditsAsync(
+            originalImage, originalImageFileName, prompt, options.Count, openAIOptions, cancellationToken).ConfigureAwait(false);
+
+        return ToTextToImageResponse(result, options);
+    }
+
+    /// <inheritdoc />
+#pragma warning disable S1067 // Expressions should not be too complex
+    public object? GetService(Type serviceType, object? serviceKey = null) =>
+        serviceType is null ? throw new ArgumentNullException(nameof(serviceType)) :
+        serviceKey is not null ? null :
+        serviceType == typeof(TextToImageClientMetadata) ? _metadata :
+        serviceType == typeof(ImageClient) ? _imageClient :
+        serviceType == typeof(OpenAIClient) ? _openAIClient :
+        serviceType.IsInstanceOfType(this) ? this :
+        null;
+#pragma warning restore S1067 // Expressions should not be too complex
+
+    /// <inheritdoc />
+    void IDisposable.Dispose()
+    {
+        // Nothing to dispose. Implementation required for the ITextToImageClient interface.
+    }
+
+    /// <summary>Converts a <see cref="GeneratedImageCollection"/> to a <see cref="TextToImageResponse"/>.</summary>
+    private static TextToImageResponse ToTextToImageResponse(GeneratedImageCollection generatedImages, TextToImageOptions options)
+    {
+        List<AIContent> contents = new();
+
+        foreach (GeneratedImage image in generatedImages)
+        {
+            if (options.ContentType == TextToImageContentType.Uri && image.ImageUri is not null)
+            {
+                contents.Add(new UriContent(image.ImageUri, "image/png"));
+            }
+            else if (options.ContentType == TextToImageContentType.Data && image.ImageBytes is not null)
+            {
+                contents.Add(new DataContent(image.ImageBytes.ToArray(), "image/png"));
+            }
+        }
+
+        return new TextToImageResponse(contents)
+        {
+            RawRepresentation = generatedImages
+        };
+    }
+
+    private ImageClient GetImageClient(TextToImageOptions options)
+    {
+        if (options.ModelId is null || options.ModelId == _defaultModelId)
+        {
+            // If default model is requested
+            return _imageClient;
+        }
+
+        // If a specific model is requested, get the image client for that model
+        return _openAIClient?.GetImageClient(options.ModelId) ??
+            throw new InvalidOperationException($"Cannot create an ImageClient for {options.ModelId}.  Please ensure {nameof(OpenAITextToImageClient)} is initialized with an {nameof(OpenAIClient)}.");
```

Comment:
Elsewhere, we just ignore properties we can't factor in, so this would just ignore the ModelId rather than throwing.